### PR TITLE
OP-124/Sylius upgrade

### PR DESCRIPTION
**Pull request by rafalmal**

### CODE REVIEW

File: composer.json.~1.12.0.dist [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
```
Comment (mpysiak) [inline composer.json.~1.12.0.dist:1]: @rafalmal I don't think this will work. 
Have you tried installing this plugin on clean Sylius 1.12?
Sylius 1.12 requires only php 8.x (it wont work on php 7.x) and symfony ^5.4 or ^6.0

**Review by mpysiak — CHANGES_REQUESTED** [review]

File: composer.json.~1.12.0.dist (hunk 1 above)
Comment (rafalmal) [reply composer.json.~1.12.0.dist:1]: Yes i tried it on sylius 1.12 with symfony 5.4, but in this case i will remove php 7.3 and 7.4 from dependencies,

Comment (mpysiak) [reply composer.json.~1.12.0.dist:1]: I think it should work on sylius 1.12 with symfony 6 also

Comment (rafalmal) [reply composer.json.~1.12.0.dist:1]: we have oauth-server package wich doesnt work with symfony 6

Comment (mpysiak) [reply composer.json.~1.12.0.dist:1]: It is not needed in sylius 1.12 anymore

Comment (mpysiak) [reply composer.json.~1.12.0.dist:1]: Please verify this, but if I remember correctly,  friendsofsymfony/oauth-server-bundle is not needed is sylius 1.11 and 1.10 also. You can remove it completely

File: composer.json [hunk 2]
```diff
@@ -26,22 +26,19 @@
...
         "phpstan/phpstan-doctrine": "0.12.39",
         "phpstan/phpstan-strict-rules": "^0.12.0",
         "phpstan/phpstan-webmozart-assert": "0.12.12",
         "phpunit/phpunit": "^9.5",
-        "sensiolabs/security-checker": "^6.0",
         "sylius-labs/coding-standard": "^4.0",
         "symfony/browser-kit": "^4.4 || ^5.2",
         "symfony/debug-bundle": "^4.4 || ^5.2",
         "symfony/dotenv": "^4.4 || ^5.2",
         "symfony/intl": "^4.4 || ^5.2",
```
Comment (mpysiak) [inline composer.json:40]: every single symfony package should be "^5.4 || ^6.0" - see sylius plugin skeleton composer - https://github.com/Sylius/PluginSkeleton/blob/1.12/composer.json

File: tests/Application/package.json.~1.11.0.dist [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (mpysiak) [inline tests/Application/package.json.~1.11.0.dist:1]: please add some info to the docs about that user have to manualy copy package.json~1.x.0.dist to package.json

eg. https://github.com/BitBagCommerce/SyliusWishlistPlugin/blob/master/doc/05-testing.md#testing

**Review by mpysiak — CHANGES_REQUESTED** [review]

**Review by mpysiak — APPROVED** [review]

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -149,6 +138,11 @@ jobs:
                     restore-keys: |
                         ${{ runner.os }}-node-${{ matrix.node }}-yarn-
 
+            -
+                name: Copy package.json.dist to package.json
```
Comment (senghe) [inline .github/workflows/build.yml:142]: We have a solution for this - aliasing in webpack, used in other PRs. Could you adjust the PR to that thing?

File: tests/Application/config/routes/sylius_shop.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ sylius_shop:
     resource: "@SyliusShopBundle/Resources/config/routing.yml"
     prefix: /{_locale}
     requirements:
-        _locale: ^[A-Za-z]{2,4}(_([A-Za-z]{4}|[0-9]{3}))?(_([A-Za-z]{2}|[0-9]{3}))?$
+        _locale: ^[a-z]{2}(?:_[A-Z]{2})?$
```
Comment (senghe) [inline tests/Application/config/routes/sylius_shop.yaml:5]: Is this change needed for Sylius 1.12?

File: tests/Application/package.json.~1.11.0.dist [hunk 6]
```diff
@@ -26,11 +26,12 @@
...
         "sass-loader": "^7.0.1",
         "upath": "^1.1.0",
+        "webpack": "^5.76.1",
         "yargs": "^6.4.0"
     },
     "scripts": {
-        "build": "gulp build",
-        "gulp": "gulp build",
+        "build": "encore production",
+        "dev": "encore dev",
```
Comment (senghe) [inline tests/Application/package.json.~1.11.0.dist:34]: Changing the build system from gulp to encore has to be mentioned in README.md file.

**Review by senghe — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 4 above)
Comment (rafalmal) [reply .github/workflows/build.yml]: Adjusted to alias resolve in next commit.

File: tests/Application/config/routes/sylius_shop.yaml (hunk 5 above)
Comment (rafalmal) [reply tests/Application/config/routes/sylius_shop.yaml]: It was not needed, restored to old value in next commit

File: tests/Application/package.json.~1.11.0.dist (hunk 6 above)
Comment (rafalmal) [reply tests/Application/package.json.~1.11.0.dist]: But thank to this change in scripts section nothing change from documentation point of view 

$ composer install
$ cd tests/Application
$ yarn install
$ yarn build // would have to be changed to yarn encore dev i i wouldnt change script section
$ bin/console assets:install public -e test
$ bin/console doctrine:database:create -e test
$ bin/console doctrine:schema:create -e test
$ bin/console server:run 127.0.0.1:8080 -d public -e test
$ sudo service elasticsearch start

Comment (senghe) [reply tests/Application/package.json.~1.11.0.dist]: :ok_hand:

**Review by senghe — APPROVED** [review]
